models/snowflake/snowplow_fractribution_sessions_by_customer_id.sql

```diff
@@ -0,0 +1,56 @@
+{{ 
+  config(
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  ) 
+}}
+
+SELECT
+    CASE
+        WHEN page_views.user_id IS NOT NULL AND page_views.user_id != '' THEN 'u' || page_views.user_id -- use event user_id
+        {% if var('use_snowplow_web_user_mapping_table') %}
+            WHEN user_mapping.domain_userid IS NOT NULL THEN 'u' || user_mapping.user_id
+        {% endif %}
+        ELSE 'f' || page_views.domain_userid
+    END AS customerId, -- f (anonymous) or u (identifier) prefixed user identifier
+    derived_tstamp AS visitStartTimestamp, -- we consider the event timestamp to be the session start, rather than the session start timestamp
+    {{ channel_classification() }} AS channel,
+    refr_urlpath AS referralPath,
+    mkt_campaign AS campaign,
+    mkt_source AS source,
+    mkt_medium AS medium
+FROM
+    {{ var('page_views_source') }}  page_views
+    {% if var('use_snowplow_web_user_mapping_table') %}
+        LEFT JOIN
+        {{ var('snowplow_web_user_mapping_table') }} AS user_mapping
+        ON
+        page_views.domain_userid = user_mapping.domain_userid
+    {% endif %}
+WHERE
+    DATE(derived_tstamp) >= DATEADD(d, -{{ var('path_lookback_days') + 1 }}, 
+                                CASE WHEN '{{ var('conversion_window_start_date') }}' = '' 
+                                THEN current_date()-31
+                                ELSE '{{ var('conversion_window_start_date') }}'
+                                END)
+    AND
+    DATE(derived_tstamp) <= CASE WHEN '{{ var('conversion_window_end_date') }}' = '' 
+                                THEN current_date()-1
+                                ELSE '{{ var('conversion_window_end_date') }}'
+                                END
+    AND
+    -- restrict to certain hostnames
+    page_urlhost IN (
+        {%- for urlhost in var('conversion_hosts') %}
+            '{{ urlhost }}'
+            {%- if not loop.last %},{% endif %}
+        {%- endfor %}
+    )
```

Comment:
If this variable is null or an empty list this will break the query - will this always have a value and you'd never want to just include all of them?

Reply:
It should be set and there is a default string they would have to overwrite with '' or null but nothing is stopping them so  I have added some exception handling just to get a proper error message if they do that.